e2e-tests/tests/infra/appSections/AddElementSection.ts

```diff
@@ -0,0 +1,58 @@
+import { Locator, Page, expect } from "@playwright/test";
+
+type ElementType = "Components" | "Containers" | "Layouts";
+
+/**
+ * A Playwright wrapper around the Add Element functionality.
+ */
+export default class AddElementSection {
+  private addElementButton: Locator;
+  private addElementMenu: Locator;
+
+  constructor(private page: Page) {
+    this.addElementButton = page.getByRole("button", {
+      name: "Open Add Element Menu",
+    });
+    this.addElementMenu = page.getByTestId("add-element-menu");
+  }
+
+  getAddElementLocator(elementName: string): Locator {
+    return this.page.getByRole("button", {
+      name: `Add ${elementName} Element`,
+    });
+  }
+
+  async toggleAddElementMenu() {
```

Comment:
nit: I think we have a similar function here for the viewport section, but we call it `openViewportMenu` or something similar.  I like that it's called `toggle` here because it seems more accurate to what it's doing.  Should we change the name in the viewport section for consistency?

Reply:
Done!